execution-plane/arcon/src/streaming/task/node.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prelude::*;
+/*
+    Node: contains Task which executes actions
+*/
+#[derive(ComponentDefinition)]
+pub struct Node<IN, OUT>
+    where
+        IN: 'static + ArconType,
+        OUT: 'static + ArconType,
+{
+    ctx: ComponentContext<Node<IN, OUT>>,
+    out_channels: Box<ChannelStrategy<OUT>>,
+    task: Box<Task<IN, OUT>>,
+}
+
+impl<IN, OUT> Node<IN, OUT>
+    where
+        IN: 'static + ArconType,
+        OUT: 'static + ArconType,
+{
+    pub fn new(
+        out_channels: Box<ChannelStrategy<OUT>>,
+        task: Box<Task<IN, OUT>>,
+    ) -> Node<IN, OUT> {
+        Node {
+            ctx: ComponentContext::new(),
+            out_channels,
+            task,
+        }
+    }
+    fn handle_event(&mut self, event: &ArconEvent<IN>) -> () {
+        match event {
+            ArconEvent::Element(e) => {
+                if let Ok(results) = self.task.handle_element(*e) {
+                    for result in results {
+                        self.output_event(result);
+                    }
+                } else {
+                    error!(self.ctx.log(), "Task failed to handle element");
+                }
+            }
+            ArconEvent::Watermark(w) => {
+                if let Ok(results) = self.task.handle_watermark(*w) {
+                    for result in results {
+                        self.output_event(result);
+                    }
+                } else {
+                    error!(self.ctx.log(), "Task failed to handle watermark");
+                }
+                self.output_event(ArconEvent::<OUT>::Watermark(*w)); // Forward the watermark
+            }
+        }
+    }
+    fn output_event(&mut self, event: ArconEvent<OUT>) -> () {
+        match self.out_channels.output(event, &self.ctx.system()) {
```

Comment:
unnecessary match statement.   doing a `if let Err(e) = .... {  error!() }` would be better